dmoj/cptbox/ptdebug.cpp

```diff
@@ -84,9 +102,67 @@ void pt_debugger::poke_reg(int idx, long data) {
 typedef int ptrace_read_t;
 #else
 typedef long ptrace_read_t;
+
+#ifndef SYS_process_vm_readv
+#define SYS_process_vm_readv 270
+#endif
+
+ssize_t __attribute__((weak)) process_vm_readv(
```

Comment:
Since this is [supposedly](http://man7.org/linux/man-pages/man2/process_vm_readv.2.html) in `sys/uio.h`, a comment as to why/what platform doesn't have it would be useful.

Reply:
Indeed. Android's bionic C doesn't have the wrapper defined even though the underlying kernel has the system call.